test/metric-end-results/TSXMetrics.test.ts

```diff
@@ -0,0 +1,49 @@
+import { expectFileMetric, parseAllFileMetrics } from "./TestHelper";
+import { FileMetric, MetricResult } from "../../src/parser/metrics/Metric";
+
+describe("TSX metrics tests.", () => {
+    const tsxTestResourcesPath = "./resources/tsx/";
+
+    let results: Map<string, Map<string, MetricResult>>;
+
+    const testFileMetric = (inputPath, metric, expected) =>
+        expectFileMetric(results, inputPath, metric, expected);
+
+    beforeAll(async () => {
+        results = await parseAllFileMetrics(tsxTestResourcesPath);
+    });
+
+    describe("parses classes metric", () => {
+        it("should count all classes, interfaces, types, enum with implementation.", () => {
+            testFileMetric(tsxTestResourcesPath + "/class-likes.tsx", FileMetric.classes, 8);
+        });
+    });
+    describe("parses functions metric", () => {
+        it("should count all function declarations and lambda expressions.", () => {
+            testFileMetric(tsxTestResourcesPath + "/functions.tsx", FileMetric.functions, 8);
+        });
+    });
+    describe("parses complexity metric", () => {
+        it("should count all if, if-else, for, while, do-while labels.", () => {
```

Comment:
We only call case, default or goto jump labels "labels"

```suggestion
        it("should count all if, if-else, for, while and do-while statements.", () => {
```